server/ot_rpc.go

```diff
@@ -152,6 +152,9 @@ func runTranscode(n *core.LivepeerNode, orchAddr string, httpc *http.Client, not
 		err = errors.New("segment / profile mismatch")
 	}
 	if err != nil {
+		if _, ok := err.(core.UnrecoverableError); ok {
+			defer panic(err)
```

Comment:
I recall that when we were pairing we saw an IDE warning about deferred panics. Curious what that warning was about and if you think it is fine to use a deferred panic here?

Reply:
Here's the whole GoLand's warning.

>defer should not call panic() directly 
> Inspection info: Reports defer and go statements that call panic() or recover() directly.
> Such statements are rarely useful and might indicate a misuse of the panic() and recover() mechanism. In particular:
> - go panic(): a newly-started goroutine will panic immediately.
> - defer panic(): a function with this statement will always panic on exit.
> - go recover(): has no effect as newly-started goroutine cannot panic.
> - defer recover(): function with this statement will silently stop panicking. This could be a valid usage, but an idiomatic way is to inspect the value returned by recover():
> ```golang 
> defer func() {
>  if r := recover(); r != nil {
>    fmt.Println("Recovered from: ", r)
>  }
> }()
> ```
> For more information about go statements and panics handling, see Handling panics and Go statements in the Go Language Specification.

So, in our case, the warning is `a function with this statement will always panic on exit`. But I think it's exactly what we want to achieve (!). Wdyt?

Reply:
Hah yeah you're right. In this case, we do want to panic, but only when the function returns so that it can execute all the logic for returning the HTTP response before the panic. Guess this is a case where we can go against the warning!